File: modules/base/types/src/main/java/candle/types/resource/PreRegisterEvent.java
package candle.types.resource;

import candle.event.CancelableEvent;

public class PreRegisterEvent<T> extends CancelableEvent<Registry<T>> {
  private final Registry<T> registry;
  private final Identifier id;

  public PreRegisterEvent( Registry<T> registry, Identifier id ) {
    super(registry);
    this.registry = registry;
    this.id = id;
  }

  public Registry<T> getRegistry() {
    return registry;
  }

  public Identifier getId() {
    return id;
  }
}

File: base/types/src/main/kotlin/net/candlemc/types/resource/PreRegisterEvent.kt
package net.candlemc.types.resource

import net.candlemc.event.CancelableEvent

class PreRegisterEvent<T>(registry: Registry<T>, val id: Identifier) :
    CancelableEvent<Registry<T>?>(registry) {
    private val registry: Registry<T> = registry

    fun getRegistry(): Registry<T> {
        return registry
    }
}
